src/error.rs

```diff
@@ -32,4 +45,56 @@ pub enum DatabaseError {
     WriteError(String),
     #[error("Failed to delete {0} from the Database database")]
     DeleteError(String),
-}
\ No newline at end of file
+}
+
+#[derive(Error, Debug)]
+pub enum DataAvailabilityError {
+    #[error("Failed to establish WebSocket connection: {0}")]
+    WebSocketError(String),
```

Comment:
I would change this to ConnectionError because not all implementations will use Websocket connections